presto-cache/src/main/java/com/facebook/presto/cache/CachingFileSystem.java

```diff
@@ -13,66 +13,88 @@
  */
 package com.facebook.presto.cache;
 
+import com.facebook.airlift.log.Logger;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.BlockLocation;
+import org.apache.hadoop.fs.ContentSummary;
+import org.apache.hadoop.fs.CreateFlag;
 import org.apache.hadoop.fs.FSDataInputStream;
 import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileAlreadyExistsException;
+import org.apache.hadoop.fs.FileChecksum;
 import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.FsServerDefaults;
+import org.apache.hadoop.fs.FsStatus;
+import org.apache.hadoop.fs.Options;
+import org.apache.hadoop.fs.ParentNotDirectoryException;
 import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.fs.UnsupportedFileSystemException;
+import org.apache.hadoop.fs.XAttrSetFlag;
+import org.apache.hadoop.fs.permission.AclEntry;
+import org.apache.hadoop.fs.permission.AclStatus;
+import org.apache.hadoop.fs.permission.FsAction;
 import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.security.AccessControlException;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.security.token.Token;
 import org.apache.hadoop.util.Progressable;
 
+import java.io.FileNotFoundException;
 import java.io.IOException;
 import java.net.URI;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Map;
 
 import static java.util.Objects.requireNonNull;
 
 public final class CachingFileSystem
         extends FileSystem
 {
-    private final URI workingDirectory;
+    private static final Logger log = Logger.get(CachingFileSystem.class);
+    private final URI uri;
     private final CacheManager cacheManager;
     private final FileSystem dataTier;
     private final boolean cacheValidationEnabled;
 
     public CachingFileSystem(
-            URI workingDirectory,
+            URI uri,
             Configuration configuration,
             CacheManager cacheManager,
             FileSystem dataTier,
             boolean cacheValidationEnabled)
     {
-        requireNonNull(workingDirectory, "uri is null");
         requireNonNull(configuration, "configuration is null");
-
-        this.workingDirectory = URI.create(workingDirectory.getScheme() + "://" + workingDirectory.getAuthority());
+        this.uri = requireNonNull(uri, "uri is null");
         this.cacheManager = requireNonNull(cacheManager, "cacheManager is null");
         this.dataTier = requireNonNull(dataTier, "dataTier is null");
         this.cacheValidationEnabled = cacheValidationEnabled;
 
         setConf(configuration);
 
         //noinspection AssignmentToSuperclassField
-        statistics = getStatistics(workingDirectory.getScheme(), getClass());
+        statistics = getStatistics(this.uri.getScheme(), getClass());
```

Comment:
remove `this.`

Reply:
This is in the constructor, and both parameter and member variable names are same. Even though both holds the same value, using this makes it clear which variable we are trying to use.